usl_models/usl_models/flood_ml/model.py

```diff
@@ -200,20 +204,22 @@ class FloodConvLSTM(tf.keras.Model):
 
     def __init__(
         self,
-        params: FloodModelParams,
+        params: dict[str, Any],
         spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
     ):
         """Creates the ConvLSTM model.
 
         Args:
-            params: A FloodModelParams object of configurable model parameters.
+            params: A dictionary representation of a FloodModelParams object for
+                configurable model parameters. We take a dictionary rather than a
+                FloodModelParams object to ensure the model is serializable.
             spatial_dims: Tuple of spatial height and width input dimensions.
                 Needed for defining input shapes. This is an optional arg that
                 can be changed (primarily for testing/debugging).
         """
         super().__init__()
 
-        self._params = params
+        self._params = model_params.FloodModelParams(**params)
```

Comment:
We need `spatial_dims` to define the input shape, but I don't consider this a hyperparameter that should be changed arbitrarily.

Another reason I thought it might be useful to define a separate dictionary for hparams is that we can write functions to return specific configurations in case we want them for future reference (not sure whether that's a likely use case here). For example, if we find `params1` and `params2` to both perform well and want to train some new models, we could store/retrieve them more easily in the code.

You could still have those functions and initialize a model using kwargs with `FloodModel(**params)`, but the model definition just feels clunkier. I've worked in code bases where the number of arguments just keeps growing and it creates quite a bit of tech debt.